docs/source/user/5minguide.rst

```diff
@@ -0,0 +1,247 @@
+from a terminal or notebook. Using the above ``loop_lift`` function, try::
+
+    print(loop_lift.inspect_types(pretty=True))
+
+
+How to measure the performance of Numba?
+----------------------------------------
+First, recall that Numba has to compile your function for the argument types
+given before it executes the machine code version of your function, this takes
+time. However, once the compilation has taken place Numba caches the machine
+code version of your function for the particular types of arguments presented.
+If it is called again the with same types, it can reuse the cached version
+instead of having to compile again.
+
+A really common mistake when measuring performance is to not account for the
+above behaviour and to time code once with a simple timer that includes the
+time taken to compile your function in the execution time.
+
+For example::
+
+    from numba import jit
+    import numpy as np
+    import time
+
+    x = np.arange(100).reshape(10, 10)
+
+    @jit
+    def go_fast(a): # Function is compiled and runs in machine code
+        trace = 0
+        for i in range(a.shape[0]):
+            trace += np.tanh(a[i, i])
+        return a + trace
+
+    # DO NOT REPORT THIS... COMPILATION TIME IS INCLUDED IN THE EXECUTION TIME!
+    start = time.time()
+    go_fast(x)
+    end = time.time()
+    print("Elapsed (with compilation) = %s" % (end - start))
+
+    # NOW THE FUNCTION IS COMPILED, RE-TIME IT EXECUTING FROM CACHE
+    start = time.time()
+    go_fast(x)
+    end = time.time()
+    print("Elapsed (after compilation) = %s" % (end - start))
+
+This, for example prints::
+
+    Elapsed (with compilation) = 0.33030009269714355
+    Elapsed (after compilation) = 6.67572021484375e-06
+
+A good way to measure the impact Numba JIT has on your code is to time execution
+using the `timeit <https://docs.python.org/3/library/timeit.html>`_ module
+functions, these measure multiple iterations of execution and, as a result,
+can be made to accommodate for the compilation time in the first execution.
+
+As a side note, if compilation time is an issue, Numba JIT supports `on-disk
+caching <LINK>`_ of compiled functions and also has an `Ahead-Of-Time <LINK>`_
+compilation mode.
+
+How fast is it?
+---------------
+Assuming Numba can operate in ``nopython`` mode, or at least compile some loops,
+it will target compilation to your specific CPU. Speed up varies depending on
+application but can be one to two orders of magnitude. Numba has a
+`performance guide <LINK>`_ that covers common options for gaining extra
+performance.
+
+How does Numba work?
+--------------------
+Glossing over a huge amount of detail, Numba reads the Python bytecode for a
+decorated function and combines this with information about the types of the
+input arguments to the function. It then forms a Numba specific internal
+representation(IR) of the function and translates this to LLVM IR, it then
```

Comment:
We can go even simpler on this discussion.  "It analyzes and optimizes your code, and finally uses the LLVM compiler library to generate a machine code version of your function, tailored to your CPU capabilities."